docfx_yaml/extension.py

```diff
@@ -943,56 +943,77 @@ def _update_friendly_package_name(path):
         if _type in [METHOD, FUNCTION, CLASS]:
             argspec = inspect.getfullargspec(obj) # noqa
             type_map = {}
-            if argspec.annotations:
-                for annotation in argspec.annotations:
-                    if annotation == "return":
-                        continue
-                    try:
-                        type_map[annotation] = _extract_type_name(
-                            argspec.annotations[annotation])
-                    except AttributeError:
-                        print(f"Could not parse argument information for {annotation}.")
-                        continue
-
-            # Add up the number of arguments. `argspec.args` contains a list of
-            # all the arguments from the function.
-            arg_count += len(argspec.args)
-            for arg in argspec.args:
-                arg_map = {}
-                # Ignore adding in entry for "self"
-                if arg != 'cls':
-                    arg_map['id'] = arg
-                    if arg in type_map:
-                        arg_map['var_type'] = type_map[arg]
-                        args.append(arg_map)
+
+            annotations = getattr(argspec, 'annotations', [])
+            for annotation in argspec.annotations:
+                if annotation == "return":
+                    continue
+                try:
+                    type_map[annotation] = _extract_type_name(
+                        argspec.annotations[annotation])
+                except AttributeError:
+                    print(f"Could not parse argument information for {annotation}.")
+                    continue
+
+            # Retrieve arguments from various sources like `argspec.args` and
+            # `argspec.kwonlyargs` for positional/keyword arguments.
+            original_args = argspec.args
+            # Stores default information for kwonly arguments. Unlike
+            # inspect.defaults, which is a tuple of default values, kw defaults
+            # is a dict[name, defaultvalue].
+            kw_defaults = {}
+            if argspec.kwonlyargs:
+                original_args.extend(argspec.kwonlyargs)
+                if argspec.kwonlydefaults:
+                    kw_defaults.update(argspec.kwonlydefaults)
+            arg_count += len(original_args)
+            for arg in original_args:
+                # Ignore adding in entry for "self" or if docstring cannot be
+                # formed for current arg, such as docstring is missing or type
```

Comment:
For my understanding, how come a type  is necessary? Can the table (like on the cgc page) just be populated with the arg name to at least indicate it exists? If this is to drop the `*` from args, I think having that be skipped instead of *all* args with no type would be prefered.